tests/README.rst

```diff
@@ -0,0 +1,76 @@
+===========
+Don't Panic
+===========
+
+The Hypothesis test suite is large, but we've written these notes to help you
+out.  It's aimed at contributors (new and old!) who know they need to add tests
+*somewhere*, but aren't sure where - or maybe need some hints on what kinds of
+tests might be useful.  Others might just be interested in how a testing
+library tests itself!
+
+
+The very short version
+======================
+
+- To improve code coverage (eg because ``make check-coverage`` / the Travis
+  build is failing), go to ``cover/``
+- For longer / system / integration tests, look in ``nocover/``
+- For tests that require an optional dependency, look in the appropriate
```

Comment:
Maybe be more explicit that "appropriate" means "with the same name as the dependency"?